storage/innobase/buf/buf0dblwr.cc

```diff
@@ -338,6 +336,92 @@ buf_dblwr_create(void)
 	goto start_again;
 }
 
+/** Compute the path to the parallel doublewrite buffer, if not already done */
+__attribute__((warn_unused_result))
+static
+dberr_t
+buf_parallel_dblwr_make_path(void)
+{
+	if (parallel_dblwr_buf.path)
+		return(DB_SUCCESS);
+
+	char path[FN_REFLEN];
+
+	if (srv_parallel_doublewrite_path
+	    && srv_parallel_doublewrite_path[0] == OS_PATH_SEPARATOR) {
+
+		strncpy(path, srv_parallel_doublewrite_path, sizeof(path));
+	} else if (srv_data_home[strlen(srv_data_home) - 1]
+		   == OS_PATH_SEPARATOR) {
+		ut_snprintf(path, sizeof(path), "%s%s", srv_data_home,
+			    srv_parallel_doublewrite_path);
+	} else {
+		ut_snprintf(path, sizeof(path), "%s%c%s", srv_data_home,
+			    OS_PATH_SEPARATOR, srv_parallel_doublewrite_path);
+	}
+
+	os_file_type_t	type;
+	bool		exists = false;
+	bool		ret;
+
+	ret = os_file_status(path, &exists, &type);
+
+	/* For realpath() to succeed the file must exist. */
+
+	if (ret && exists) {
+		if (my_realpath(path, path, MY_WME) != 0)
+			return(DB_ERROR);
+		if (type != OS_FILE_TYPE_FILE) {
+			ib::error() << "Parallel doublewrite path "
+				    << path << " must point to a regular "
+				"file";
+			return(DB_WRONG_FILE_NAME);
+		}
+	} else if (srv_parallel_doublewrite_path[0] != OS_PATH_SEPARATOR) {
+		/* If it does not exist, and is not an absolute path, then
+		resolve only srv_data_home and append
+		srv_parallel_doublewrite_path to it. */
+		char	srv_data_home_full[FN_REFLEN];
+
+		if (my_realpath(srv_data_home_full, srv_data_home, MY_WME)
+		    != 0) {
+			return(DB_ERROR);
+		}
+
+		if (srv_data_home_full[strlen(srv_data_home_full) - 1]
```

Comment:
I don't know: http://pubs.opengroup.org/onlinepubs/9699919799/functions/realpath.html and http://man7.org/linux/man-pages/man3/realpath.3.html are not clear to me with this respect.